service-app/internal/api/service_documents_test.go

```diff
@@ -13,76 +13,100 @@ import (
-			EmbeddedXML: xmlData,
-			EmbeddedPDF: base64.StdEncoding.EncodeToString(data),
-			Type:        "Correspondence",
-		},
-		set: &types.BaseSet{
-			Header: &types.BaseHeader{
-				ScanTime: "2024-12-05 12:34:56Z",
+	// Set up expected interactions
+	mockProvider.
+		AddInteraction().
+		Given("An LPA with UID 7000-3764-4871 exists").
+		Given("I am a DDC user").
+		UponReceiving("A request to attach a scanned document").
+		WithRequest("POST", "/api/public/v1/scanned-documents", func(b *consumer.V4RequestBuilder) {
+			b.
+				Header("Content-Type", matchers.String("application/json")).
+				JSONBody(matchers.Map{
+					"caseReference":   matchers.String("7000-3764-4871"),
+					"content":         matchers.String(pdfBase64),
+					"documentType":    matchers.String("Correspondence"),
+					"documentSubType": matchers.String("Legal"),
+					"scannedDate":     matchers.DateTimeGenerated("2025-02-16T11:40:45Z", "yyyy-MM-dd'T'HH:mm:ss'Z'"),
+				})
+		}).
+		WillRespondWith(201, func(b *consumer.V4ResponseBuilder) {
+			b.
+				Header("Content-Type", matchers.String("application/json")).
+				JSONBody(matchers.Map{
+					"uuid": matchers.UUID(),
+				})
+		})
+
+	err = mockProvider.ExecuteTest(t, func(pactConfig consumer.MockServerConfig) error {
+		baseURL := fmt.Sprintf("http://%s:%d", pactConfig.Host, pactConfig.Port)
+
+		// Mock dependencies
+		mockConfig := config.NewConfig()
+		mockConfig.App.SiriusBaseURL = baseURL
+		logger := logger.NewLogger(mockConfig)
+
+		_, _, _, tokenGenerator := auth.PrepareMocks(mockConfig, logger)
+		httpClient := httpclient.NewHttpClient(*mockConfig, *logger)
+		httpMiddleware, _ := httpclient.NewMiddleware(httpClient, tokenGenerator)
+
+		// Prepare service instance
+		service := &Service{
+			Client: &Client{Middleware: httpMiddleware},
+			originalDoc: &types.BaseDocument{
+				EmbeddedXML: xmlBase64,
+				EmbeddedPDF: pdfBase64,
+				Type:        "Correspondence",
 			},
-		},
-	}
+			set: &types.BaseSet{
+				Header: &types.BaseHeader{
+					ScanTime: "2024-12-05 12:34:56Z",
+				},
+			},
+		}
 
-	caseResponse := &types.ScannedCaseResponse{
-		UID: "CASE123",
-	}
+		caseResponse := &types.ScannedCaseResponse{
+			UID: "7000-3764-4871",
+		}
 
-	// Mock the response from HTTPRequest
-	mockResponse := &types.ScannedDocumentResponse{
-		UUID:                "CASE123",
-		Type:                "Correspondence",
-		Subtype:             "Application Related",
-		SourceDocumentType:  "Correspondence",
-		Title:               "Case Document",
-		FriendlyDescription: "Scanned document for case",
-		ID:                  1,
-	}
-	mockResponseBytes, _ := json.Marshal(mockResponse)
+		ctx := context.Background()
+		response, decodedXML, err := service.AttachDocuments(ctx, caseResponse)
+		if err != nil {
+			t.Fatalf("AttachDocuments returned error: %v", err)
+		}
+		assert.NotNil(t, response)
+		assert.Equal(t, xmlStringData, decodedXML)
+		assert.Equal(t, "fc763eba-0905-41c5-a27f-3934ab26786c", response.UUID)
```

Comment:
Will UUID change on future requests? Based on `matchers.UUID()` above. Should the UUID- not be dynamic and therefore we could use regex to check the pattern instead? e.g. `assert.Regexp(t, `^[uuid-pattern]`, response.UUID, "Response UUID format is incorrect")`

Reply:
I noticed in scanning-sirius.json its hard-coded, I'll assume its unlikely to change and therefore we can leave it as-is.

Reply:
`matchers.UUID()` returns a hard-coded result of `fc763eba-0905-41c5-a27f-3934ab26786c`. In the contract though, it adds a matcher to check it's a UUID with regex. So when Sirius runs the test it can come back with a random value, as long as it's a UUID.